contracts/HATVault.sol

```diff
@@ -0,0 +1,1092 @@
+        committee = _params.committee;
+        depositPause = _params.isPaused;
+        registry = _registry;
+        __ReentrancyGuard_init();
+        _transferOwnership(_params.owner);
+        tokenLockFactory = _registry.tokenLockFactory();
+
+        // Set vault to use default registry values where applicable
+        hatBountySplit = HATVaultsRegistry.HATBountySplit({
+            governanceHat: NULL_UINT,
+            hackerHatVested: 0
+        });
+        arbitrator = NULL_ADDRESS;
+        challengePeriod = NULL_UINT;
+        challengeTimeOutPeriod = NULL_UINT;
+    }
+
+    /* ---------------------------------- Claim --------------------------------------- */
+
+    /**
+    * @notice Called by the committee to submit a claim for a bounty payout.
+    * This function should be called only on a safety period, when withdrawals
+    * are disabled.
+    * Upon a call to this function by the committee the vault's withdrawals
+    * will be disabled until the claim is approved or dismissed. Also from the
+    * time of this call the arbitrator will have a period of 
+    *`HATVaultsRegistry.challengePeriod` to challenge the claim.
+    * @param _beneficiary The submitted claim's beneficiary
+    * @param _bountyPercentage The submitted claim's bug requested reward percentage
+    */
+    function submitClaim(address _beneficiary, uint256 _bountyPercentage, string calldata _descriptionHash)
+    external
+    onlyCommittee
+    noActiveClaim
+    notEmergencyPaused
+    returns (bytes32 claimId)
+    {
+        HATVaultsRegistry.GeneralParameters memory generalParameters = registry.getGeneralParameters();
+        // require we are in safetyPeriod
+        // solhint-disable-next-line not-rely-on-time
+        if (block.timestamp % (generalParameters.withdrawPeriod + generalParameters.safetyPeriod) <
+        generalParameters.withdrawPeriod) revert NotSafetyPeriod();
+        if (_bountyPercentage > maxBounty)
+            revert BountyPercentageHigherThanMaxBounty();
+        claimId = keccak256(abi.encodePacked(address(this), nonce++));
+        activeClaim = Claim({
+            claimId: claimId,
+            beneficiary: _beneficiary,
+            bountyPercentage: _bountyPercentage,
+            committee: msg.sender,
+            // solhint-disable-next-line not-rely-on-time
+            createdAt: block.timestamp,
+            challengedAt: 0
+        });
+
+        emit SubmitClaim(
+            claimId,
+            msg.sender,
+            _beneficiary,
+            _bountyPercentage,
+            _descriptionHash
+        );
+    }
+
+    /**
+    * @notice Called by the arbitrator to challenge a claim for a bounty
+    * payout that had been previously submitted by the committee.
+    * Can only be called during the challenge period after submission of the
+    * claim.
+    * @param _claimId The claim ID
+    */
+    function challengeClaim(bytes32 _claimId) external onlyArbitrator isActiveClaim(_claimId) {
```

Comment:
We should also check here that the claim has not been challenged yet

Reply:
addressing this here: https://github.com/hats-finance/hats-contracts/pull/317

Reply:
Good catch :)